### PR TITLE
add the option of upsample function for tiny vae

### PR DESCRIPTION
**Pull request by IDKiro**
# What does this PR do?

Fixes # (issue)

1. Provide other options for the upsampling function of the Tiny VAE's decoder (default is nearest), which will be used in [SDXS](https://github.com/IDKiro/sdxs).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/models/autoencoders/vae.py [hunk 1]
```diff
@@ -926,6 +926,7 @@ def __init__(
         block_out_channels: Tuple[int, ...],
         upsampling_scaling_factor: int,
         act_fn: str,
+        upsample_fn: str,
```
Comment (sayakpaul) [inline src/diffusers/models/autoencoders/vae.py:929]: We need to have the default of `Upsample` here.

File: src/diffusers/models/autoencoders/autoencoder_tiny.py [hunk 2]
```diff
@@ -102,6 +102,7 @@ def __init__(
         encoder_block_out_channels: Tuple[int, ...] = (64, 64, 64, 64),
         decoder_block_out_channels: Tuple[int, ...] = (64, 64, 64, 64),
         act_fn: str = "relu",
+        upsample_fn: str = "nearest",
```
Comment (sayakpaul) [inline src/diffusers/models/autoencoders/autoencoder_tiny.py:105]: Is it the original default value of the function associated?

**Review by sayakpaul — COMMENTED** [review]
I am cool with it. Out of curiosity: did you benefit from using a non-default upsampler?

**Review by yiyixuxu — APPROVED** [review]
I like this! thank you!

**Comment by sayakpaul** [comment]
@IDKiro once my comments are addressed, happy to merge the PR.

**Comment by IDKiro** [comment]
> @IDKiro once my comments are addressed, happy to merge the PR.

Demonstrate a special sample (most samples require a closer look) where images are generated using the model.
Use nearest:
image
Using bilinear:
image

File: src/diffusers/models/autoencoders/autoencoder_tiny.py (hunk 2 above)
Comment (IDKiro) [reply src/diffusers/models/autoencoders/autoencoder_tiny.py:105]: yes

**Comment by sayakpaul** [comment]
Just have a single comment to be addressed and then we're good to go: https://github.com/huggingface/diffusers/pull/7604#discussion_r1556719395.

File: src/diffusers/models/autoencoders/vae.py (hunk 1 above)
Comment (IDKiro) [reply src/diffusers/models/autoencoders/vae.py:929]: Maybe this is not necessary. I checked the code, DecoderTiny will only be called when AutoencoderTiny is initialized, and upsample_fn is bound to pass parameters, consistent with act_fn. The default value of nn.Upsample is "nearest", which upsample_fn will pass. So the change will not affect the original TAESD.

Comment (sayakpaul) [reply src/diffusers/models/autoencoders/vae.py:929]: Okay.

**Comment by redouanelg** [comment]
It would be nice to be able to choose this possibility for `AutoencoderKL` too

**Pull request merged**